src/TrackPanel.cpp

```diff
@@ -7795,8 +7806,20 @@ void TrackPanel::OnTrackMenu(Track *t)
    }
 
 #if defined(USE_MIDI)
-   if (t->GetKind() == Track::Note)
+   if (t->GetKind() == Track::Note) {
       theMenu = mNoteTrackMenu.get();
+      NoteTrack *track = (NoteTrack *)t;
+
+      auto & channelList = mMidiChanMenu->GetMenuItems();
+
+      for (auto item : channelList) {
+         auto id = item->GetId();
+         if (id >= OnFirstToggleChannelID && id <= OnLastToggleChannelID) {
+            auto channel = id - OnFirstToggleChannelID;
+            theMenu->Check( id, track->IsVisibleChan(channel) );
```

Comment:
I use `Check` both for enabling and disabling here (unlike what is done within `SetMenuCheck`) because I presume that `SetMenuCheck`'s message ("We only need to set check marks. Clearing checks causes problems on Linux (bug 851)") is related to the other menus being radio buttons, while I'm using check boxes.  However, since I'm not running Linux, I may be wrong about this (although testing on windows there does not seem to be issues with modifications affecting multiple tracks).